### PR TITLE
Release 0.7.0

### LINKED ISSUE #2465: Release 0.7.0
I would like to make 0.7.0 release within 1-2 weeks.

0.7.0 instead of 0.6.1 - mainly due to drop support to integration.json.

For now, I see only one blocker for this #2452

All 0.7.0 issues: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues?q=is%3Aopen+is%3Aissue+milestone%3A0.7.0

### PR DESCRIPTION
**Pull request by Kielek**
## Why

Fixes #2465

## What

Release 0.7.0.
Skipped part of config.md modification. Metrics documentation for packages link still to main due to lack of releases of packages.

## Tests

* [x] CI, except the demo due to lack of 0.7.0 release.
* [x] MacOS with Linux Containers
* [x] Windows with Linux Containers

## Checklist

- [x] `CHANGELOG.md` is updated.
- [x] Documentation is updated.
- ~~[ ] New features are covered by tests.~~

### CODE REVIEW

**Comment by pellared** [comment]
MacOS with Linux Containers **PASSED**

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,7 +5,21 @@ All notable changes to this component are documented in this file.
...
+
+### Deprecated
+
+### Removed
+
+### Fixed
+
+### Security
+
+## [0.7.0](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/releases/tag/v0.7.0)
```
Comment (pellared) [inline CHANGELOG.md:22]: A lot has been added. Maybe it is worth to add some description?

I think it is worth to call out that:

1. Work towards stabilization
2. Fallback for the service name.
3. Added support for `OTEL_LOG_LEVEL` and improved logging
4. Resource detectors and configuration 
5. Improved user experience (thanks to the rule engine).

**Review by pellared — APPROVED** [review]

**Review by nrcventura — APPROVED** [review]

**Review by pjanotti — APPROVED** [review]

**Comment by pjanotti** [comment]
`demo / build (pull_request)` is expected since it attempts to download and run using v0.7.0 that was not published yet.

File: CHANGELOG.md (hunk 1 above)
Comment (Kielek) [reply CHANGELOG.md:22]: Most of them are in added section.

I would like to keep it to communicate new dependencies on OTel/DiagnostiSource

**Pull request merged**